azurerm/internal/services/maintenance/tests/resource_arm_maintenance_configuration_test.go

```diff
@@ -0,0 +1,232 @@
+		if rs.Type != "azurerm_maintenance_configuration" {
+			continue
+		}
+
+		id, err := parse.MaintenanceConfigurationID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := conn.Get(ctx, id.ResourceGroup, id.Name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testCheckAzureRMMaintenanceConfigurationExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := acceptance.AzureProvider.Meta().(*clients.Client).Maintenance.ConfigurationsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		id, err := parse.MaintenanceConfigurationID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := conn.Get(ctx, id.ResourceGroup, id.Name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on maintenanceConfigurationsClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Maintenance Configuration %q (resource group: %q) does not exist", id.Name, id.ResourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMMaintenanceConfiguration_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_maintenance_configuration" "test" {
+  name                = "acctest-mc%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  scope               = "All"
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func testAccAzureRMMaintenanceConfiguration_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMMaintenanceConfiguration_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_maintenance_configuration" "import" {
+  name                = azurerm_maintenance_configuration.test.name
+  resource_group_name = azurerm_maintenance_configuration.test.resource_group_name
+  location            = azurerm_maintenance_configuration.test.location
+  scope               = azurerm_maintenance_configuration.test.scope
+}
+`, template)
+}
+
+func testAccAzureRMMaintenanceConfiguration_complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_maintenance_configuration" "test" {
+  name                = "acctest-mc%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  scope               = "Host"
+
+  tags = {
+    env = "test"
```

Comment:
could we ensure uppercase works?
```suggestion
    EnV = "TesT"
```

Reply:
I have changed the value to `TesT`
as for the key, this service will return small case key, regardless of the input case.
If we set `EnV = "TesT"`, the service will return `env = "TesT"`